src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -2640,22 +2640,28 @@ private string GetCustomMarkupExtensionValue(XamlMemberDefinition member)
 
 			// Get the attribute from the custom markup extension class then get the return type specifed with MarkupExtensionReturnTypeAttribute
 			var attributeData = markupType.FindAttribute(XamlConstants.Types.MarkupExtensionReturnTypeAttribute);
+			var returnType = attributeData?.NamedArguments.FirstOrDefault(kvp => kvp.Key == "ReturnType").Value.Value;
+			var cast = string.Empty;
 
-			if (attributeData == null)
+			// If we cannot determine the return type to cast with for the markup extension,
+			// we then try to get the type of the target property
+			if (returnType == null)
 			{
-				this.Log().Error($"The custom markup extension {markupTypeDef.Type.Name} must specify the return type using {nameof(XamlConstants.Types.MarkupExtensionReturnTypeAttribute)}.");
-				return string.Empty;
+				var propertyType = FindPropertyType(member.Member);
+				cast = GetCastString(propertyType, null);
+			}
+			else
+			{
+				cast = $"({returnType})";
 			}
 
-			var returnType = attributeData.NamedArguments.FirstOrDefault(kvp => kvp.Key == "ReturnType").Value.Value;
-
-			if (returnType == null)
+			if (string.IsNullOrEmpty(cast))
 			{
-				this.Log().Error($"The custom markup extension contains a ReturnType that cannot be found.");
+				this.Log().Error($"Unable to determine the return type needed for the markup extension.");
```

Comment:
```suggestion
				this.Log().Error($"Unable to determine the return type needed for the markup extension (A MarkupExtensionReturnType attribute is not available, and {member.Member} cannot be found).");
```

Reply:
I'll be pushing a commit soon which does away with that log 👍

Reply:
done